aea/cli/test.py

```diff
@@ -255,6 +264,63 @@ def packages(click_context: click.Context, author: Tuple[str]) -> None:
         sys.exit(1)
 
 
+class CoverageContext:
+    """Coveragerc file context"""
+
+    def __init__(
+        self,
+        root_dir: Path,
+        append: bool = False,
+    ) -> None:
+        """Initialize object."""
+
+        self.root_dir = root_dir
+        self.append = append
+
+        self._t = tempfile.TemporaryDirectory()
+        self.coveragerc_file = Path(self._t.name, COVERAGERC_FILE)
+
+    def pytest_args(
+        self,
+        test_path: Path,
+    ) -> List[str]:
+        """Returns coverage flags for pytest command"""
+
+        args = [
+            f"--cov-config={self.coveragerc_file}",
+            "--cov-report=term",
+            f"--cov={test_path}",
+        ]
+
+        if self.append:
+            args.append("--cov-append")
+
+        return args
+
+    def __enter__(
+        self,
+    ) -> "CoverageContext":
+        """Enter context."""
+        self.coveragerc_file.write_text(
+            COVERAGERC_CONFIG.format(root_dir=self.root_dir),
+        )
+        return self
+
+    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+        """Exit context."""
+
+        with contextlib.suppress(OSError, PermissionError):
+            self._t.cleanup()
```

Comment:
I suggest using contextlib.contextmanager for this so you can simplify writing context managers in the future.